app/service/dog.js

```diff
@@ -1,46 +1,70 @@
 const { sendUpdateToAudit } = require('../messaging/send-audit')
 const { DOG } = require('../constants/event/audit-event-object-types')
 const { statuses } = require('../constants/statuses')
+const { mapDogDaoToDog } = require('../repos/mappers/cdo')
 
 class DogService {
   constructor (dogRepository, breachesRepository) {
     this._dogRepository = dogRepository
     this._breachesRepository = breachesRepository
   }
 
-  async setBreaches (dogIndex, dogBreaches, user) {
-    /**
-     * @type {import('../data/domain/dog')}
-     */
-    const dog = await this._dogRepository.getDogModel(dogIndex)
-    const allDogBreaches = await this._breachesRepository.getBreachCategories()
-
-    /**
-     * @param {BreachCategory} breach
-     * @return {string}
-     */
-    const mapBreach = (breach) => breach.label
-
+  /**
+   *
+   * @param {Dog} dog
+   * @param {string[]} dogBreaches
+   * @param {BreachCategory[]} allDogBreaches
+   * @param user
+   * @private
+   */
+  _prepareBreaches (dog, dogBreaches, allDogBreaches, user) {
     const preAuditDog = {
```

Comment:
Will the preAuditDog always be blank in terms of breaches? Could we have a situation where a dog's breaches change from one list to another?

Reply:
this was the reason for our call last week and the question to Ben.  Theoretically if someone calls this endpoint with an In Breach dog the api allows you to set to In Breach, but update the breach reasons.  When that happens the audit log shows only changed fields.  As it was decided that the audit log should only show as describe (set to In breach, with the reasons), this approach creates the audit log in that fashion.